src/chainbin/nnet3-chain-get-egs.cc

```diff
@@ -200,6 +231,16 @@ int main(int argc, char *argv[]) {
     po.Register("srand", &srand_seed, "Seed for random number generator ");
     po.Register("length-tolerance", &length_tolerance, "Tolerance for "
                 "difference in num-frames between feat and ivector matrices");
+    po.Register("supervision-length-tolerance", &supervision_length_tolerance, "Tolerance for "
+                "difference in num-frames-subsampled between supervision and deriv weights");
+    po.Register("deriv-weights-rspecifier", &deriv_weights_rspecifier,
+                "Per-frame weights (only binary - 0 or 1) that specifies "
+                "whether a frame's gradient must be backpropagated or not. "
+                "Not specifying this is equivalent to specifying a vector of "
+                "all 1s.");
+    po.Register("normalization-scale", &scale, "Scale the weights from the "
```

Comment:
Rename  the variable 'scale' to 'normalization_fst_scale' for clarity.
And add: "(useful for unsupervised training)."